step-grid-parent/step-grid-server/src/test/java/step/grid/GridImplTest.java

```diff
@@ -52,6 +65,22 @@ public void testCleanup() throws Exception {
 		Assert.assertTrue(testFile.delete());
 	}
 
+	@Test
+	public void testCleanupNonCleanable() throws Exception {
+		GridImpl grid = getGridForCleanupTest();
+		grid.start();
+
+		File testFile = File.createTempFile("GridImplTest",".txt");
+
+		FileVersion version = grid.registerFile(testFile, false);
+
+		grid.cleanupFileManagerCache();
+
+		Assert.assertEquals(version, grid.getRegisteredFile(version.getVersionId()));
+
+		Assert.assertTrue(testFile.delete());
+	}
+

```

Comment:
Not sure if we test the scheduler that performs the deletion. We might add a test to cover this.